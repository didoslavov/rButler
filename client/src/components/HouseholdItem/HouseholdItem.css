.household {
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
    width: 600px;
    gap: 20px;
    box-shadow: 5px 5px 10px 5px rgba(45, 71, 90, 0.5);
    border-radius: 4px;
    padding: 30px;
    position: relative;
    transition: color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, background-color 550ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.household:hover {
    background-color: var(--dark-blue);
    color: var(--light-grey);
}

.chip-container {
    position: absolute;
    top: 10px;
    right: 10px;
    z-index: 1;
}
