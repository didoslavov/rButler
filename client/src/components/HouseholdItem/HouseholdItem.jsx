import { Chip } from '@mui/material';
import { chipStyles } from '../../styles/muiStyles/muiStyles.js';

const HouseholdItem = ({ household }) => {
    return (
        <li className="household">
            <div className="chip-container">
                <Chip sx={chipStyles} label={'Owner: ' + household.master.username} />
            </div>
            <h4 className="border-bottom">{household.name}</h4>
            <p>{household.presentation}</p>
        </li>
    );
};

export default HouseholdItem;
