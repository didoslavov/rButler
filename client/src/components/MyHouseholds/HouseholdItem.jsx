import { Chip } from '@mui/material';
import { chipStyles } from '../../styles/muiStyles/muiStyles.js';

const HouseholdItem = ({ household, isMaster }) => {
    return (
        <li className="household">
            <div className="chip-container">
                <Chip sx={chipStyles} label={isMaster ? 'master' : 'resident'} />
            </div>
            <h4 className="household-header border-bottom">{household.name}</h4>
            <p>{household.presentation}</p>
        </li>
    );
};

export default HouseholdItem;
