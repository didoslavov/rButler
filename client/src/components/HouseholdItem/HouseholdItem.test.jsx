import React from 'react';
import { render, screen } from '@testing-library/react';
import HouseholdItem from './HouseholdItem';

const mockHousehold = {
    name: 'Test Household',
    presentation: 'A wonderful place',
    master: {
        username: 'testuser',
    },
};

describe('HouseholdItem Component', () => {
    it('renders household name', () => {
        render(<HouseholdItem household={mockHousehold} />);
        expect(screen.getByText('Test Household')).toBeInTheDocument();
    });

    it('renders household presentation', () => {
        render(<HouseholdItem household={mockHousehold} />);
        expect(screen.getByText('A wonderful place')).toBeInTheDocument();
    });

    it('renders owner username', () => {
        render(<HouseholdItem household={mockHousehold} />);
        expect(screen.getByText('Owner: testuser')).toBeInTheDocument();
    });

    it('does not render a non-existent property', () => {
        render(<HouseholdItem household={{ name: 'Test Household' }} />);
        expect(screen.queryByText('Owner:')).toBeNull();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<HouseholdItem household={mockHousehold} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
