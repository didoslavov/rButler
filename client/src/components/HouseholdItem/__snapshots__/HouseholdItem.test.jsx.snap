// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HouseholdItem Component > matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="household"
  >
    <div
      class="chip-container"
    >
      <div
        class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-s8zvmw-MuiChip-root"
      >
        <span
          class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
        >
          Owner: testuser
        </span>
      </div>
    </div>
    <h4
      class="border-bottom"
    >
      Test Household
    </h4>
    <p>
      A wonderful place
    </p>
  </li>
</DocumentFragment>
`;
